python/kserve/kserve/handlers/dataplane.py

```diff
@@ -233,7 +235,7 @@ async def infer(
             is_cloudevent = True
             is_binary_cloudevent = True
             body = self.get_binary_cloudevent(body, headers)
-        else:
+        elif headers["content-type"] in JSON_HEADERS or headers["Content-Type"] in JSON_HEADERS:
```

Comment:
@SatishReddyBethi see the test failures here https://github.com/kserve/kserve/actions/runs/3502614095/jobs/5950435196, we need to add None check for `headers`

Reply:
@yuzisun I have added the check. Can you rerun the test?